### ISSUE TITLE
Allow users to remove primary key constraints

### ISSUE THREAD

**Issue by pavish** [centerofci/mathesar#2686]
From https://github.com/centerofci/mathesar/issues/439:

We should allow users to remove the primary key constraint for a column. If a column has no primary key, then it loses a lot of functionality and we need to warn the users about that.

This issue _might_ need some design work for warning the users about the effects of removing the primary key. Whoever picks the issue up should come up with a general design and get in touch with the design team (cc: @ghislaineguerin) to get a confirmation.

Blocked by:
- https://github.com/centerofci/mathesar/issues/2685
- https://github.com/centerofci/mathesar/issues/2245

**Comment by zackkrida** [comment]
Removing this from the v0.2.2 milestone as we've decided not to allow full control of PKs from the table view. For now we're only going to allow configuring PKs at import and table creation, as part of the #4310 project.

**Comment by kgodey** [comment]
If anyone needs this in Mathesar, please create a new issue! For now I've closed as not planned since this was superseded by the decisions in #4310.
